src/containers/Tasks/RequestTerms/RequestTerms.tsx

```diff
@@ -47,35 +43,36 @@ class RequestLegitimation extends React.Component<
     }
   }
 
-  private isValid(): number {
-    const { receiverAddresses } = this.props
-    return receiverAddresses && receiverAddresses.length
-  }
-
   private handleSubmit(): void {
     const { cTypeHash, receiverAddresses, onFinished } = this.props
     const { selectedClaimEntries } = this.state
-    let claims: sdk.IPartialClaim[] = cTypeHash ? [{ cTypeHash }] : []
+    let claims: sdk.IPartialClaim[] = [{ cTypeHash } as sdk.IPartialClaim]
```

Comment:
Because CtypeHash is optional, this would potentially allow sending requests with partial claims that do not have ctypeHashes - does that make sense?

Reply:
Timo doesn't want us to use the `as ....[some type]` 

I understand that maybe the following instead. 
` if (cTypeHash) {
      let claims: sdk.IPartialClaim[] = [{ cTypeHash }]

      if (selectedClaimEntries && selectedClaimEntries.length) {
        claims = selectedClaimEntries.map(
          (claimEntry: Claims.Entry) => claimEntry.claim
        )
      }

      if (this.isValid()) {
        attestationWorkflow.requestTerms(claims, receiverAddresses).then(() => {
          if (onFinished) {
            onFinished()
          }
        })
      }
    }`

Reply:
``if (cTypeHash) {
      let claims: sdk.IPartialClaim[] = [{ cTypeHash }]
....
    }``

Reply:
sure, combined with an appropriate error message asking to specify a ctype hash, this would work nicely

Reply:
Added an error. Let me know if you like it.